Query cleanup

smitpatel opened the pull request:
Most of changes are renames and moving files to Internal folder. I will comment on the breaking changes for visibility.

smitpatel commented on src/EFCore/Query/ExpressionPrinter.cs:
```diff
@@ -51,8 +51,6 @@ public ExpressionPrinter()
             _encounteredParameters = new List<ParameterExpression>();
         }
 
-        public virtual IndentedStringBuilder StringBuilder => _stringBuilder;
```
Breaking change. Use API on expression printer directly

smitpatel commented on src/EFCore/Query/ExpressionPrinter.cs:
```diff
@@ -101,18 +99,6 @@ public virtual ExpressionPrinter AppendLines([NotNull] object o, bool skipFinalN
             return this;
         }
 
-        public virtual ExpressionPrinter IncrementIndent()
```
Remove Increment/Decrement Indent. Using Indent and disposable pattern.

smitpatel commented:
>     public class QueryOptimizerFactory : IQueryOptimizerFactory

Moved to Internal as it is factory.

---
Refers to: src/EFCore/Query/QueryOptimizerFactory.cs:18 in d309c50. [](commit_id = d309c503034fe19fc653bba0d03d3b6d55a434e1, deletion_comment = True)

ajcvickers commented:
@smitpatel I intentionally made the factory public because we have dependencies for it, so it makes sense for providers to extend from our factory.

smitpatel commented on src/EFCore.Relational/Query/RelationalShapedQueryOptimizer.cs:
```diff
@@ -32,17 +35,19 @@ public override Expression Visit(Expression query)
             return query;
         }
+
+        protected virtual Expression OptimizeSqlExpression(Expression query) => _sqlExpressionOptimizingExpressionVisitor.Visit(query);
```
All visitors in optimizers are supposed to be internal and exposed via protected methods. This particular visitor moved to .Internal but being used by SqlServer so I have added method to facilitate this for now. We can go over and add methods to call other visitors too in future.

smitpatel commented:
@ajcvickers - I will check and revert if necessary.

smitpatel commented on src/EFCore.Relational/Query/RelationalQueryContextFactory.cs:
```diff
@@ -38,15 +38,6 @@ public class RelationalQueryContextFactory : QueryContextFactory
-        /// <value>
-        ///     The execution strategy factory.
-        /// </value>
-        protected virtual IExecutionStrategyFactory ExecutionStrategyFactory
```
Factory should not be exposing anything from Dependency object.

smitpatel commented on src/EFCore/Query/IParameterValues.cs:
```diff
@@ -11,9 +11,5 @@ public interface IParameterValues
 
         void Add([NotNull] string name, [CanBeNull] object value);
-
-        object Remove([NotNull] string name);
```
These APIs were used in old pipeline. Not anymore. It should not affect any provider.

smitpatel commented on src/EFCore/Query/Internal/AnonymousObject.cs:
```diff
@@ -117,21 +81,5 @@ public override int GetHashCode()
-        ///     any release. You should only use it directly in your code with extreme caution and knowing that
-        ///     doing so can result in application failures when updating to a new Entity Framework Core release.
-        /// </summary>
-        public object GetValue(int index) => _values[index];
```
This was to support include in old pipeline. No longer needed.

smitpatel commented:
Added another commit to 
- Remove QueryContextFactory. (Was abstract class which did no specific computation).
- Providers implement interface directly. Relational factory was also moved to internal. This is provider breaking change. Npgsql should work since it derives from Relational.

smitpatel commented:
We will go through breaking changes in API review again. 🎉
